R/git_tools.R

```diff
@@ -48,9 +48,39 @@ checkout_repo <- function(repo_dir, repo_url, select_branch_rule, token_envvar,
       message(paste("clone", repo_url, "to directory", repo_dir))
     }
 
-    git_repo <- git2r::clone(
-      url = repo_url, local_path = repo_dir, credentials = creds, progress = verbose >= 2
-    )
+    tryCatch({
+      git_repo <- git2r::clone(
+        url = repo_url, local_path = repo_dir, credentials = creds, progress = verbose >= 2
+      )
+    }, error = function(e) {
+      # catch some common errors
+      # only do this when cloning because the API calls introduce quite some time overhead
+      if (grepl("https://github.com", repo_url, fixed = TRUE)) {
```

Comment:
I assume we're not trying to do something similar for gitlab?

Reply:
I did this now for Roche gitlab. I also figured out how to pass the authorization, so we don't depend on the `gh` package.